packages/core/src/electron-main/electron-main-menu.ts

```diff
@@ -0,0 +1,238 @@
+// *****************************************************************************
+// Copyright (C) 2022 Arduino SA and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { injectable } from 'inversify';
+import { app, BrowserWindow, ipcMain, WebContents } from '../../electron-shared/electron';
+import { Menu } from '../../electron-shared/electron/index';
+import { isOSX } from '../common';
+import { MenuItemConstructorOptions } from '../electron-common/menu';
+import {
+    CloseContextMenu,
+    ContextMenuDidClose, MenuItemDidClick, SetMenu, ShowContextMenu, UpdateMenuItems
+} from '../electron-common/messaging/electron-messages';
+import { ElectronMainApplicationContribution } from './electron-main-application';
+
+@injectable()
+export class ElectronMainMenu implements ElectronMainApplicationContribution {
```

Comment:
`ElectronMainMenu` is a bit ambiguous, esp. since this also handles context menus. Perhaps `ElectronMenuHandler` on `ElectronApplicationMenuHandler`?

Reply:
I see your point; however, appending `Handler`, `Manager`, or `Service`, .etc does not clarify much. Please pick a name, and I will use that in the next PR update. Thanks!